packages/data-stores/src/queries/use-jetpack-search-ai.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+
+type AIResponseURL = {
+	url: string;
+	title: string;
+};
+
+export type JetpackSearchAIResult = {
+	response: string;
+	step: string;
+	urls: AIResponseURL[];
+	terms: string[];
+};
+
+export function useJetpackSearchAIQuery( siteId: number | string, query: string, stopAt: string ) {
+	return useQuery< JetpackSearchAIResult >(
+		[ query ],
+		async () =>
+			await wpcomRequest( {
+				path: `sites/${ siteId }/jetpack-search/ai/search`,
+				apiNamespace: 'wpcom/v2/',
+				apiVersion: '2',
+				query: `query=${ encodeURIComponent( query ) }&stop_at=${ stopAt }`,
+			} ),
```

Comment:
This won't work on Atomic sites. If we already have a Jetpack endpoint for search, we can use it like [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/help-center/src/hooks/use-help-search-query.ts#L21-L31).

Reply:
Yes sir, `wpcomRequest` actively blocks requests from Atomic sites. If this API is public, you can use `apiFetch` instead, which will work everywhere if CORS is allowed.

Reply:
Implemented, not tested yet.